### PR TITLE
Validate that the PO Number is set on the payment instance.

### PR DESCRIPTION
**Pull request by centerax**
### Description
Validate that the `po_number` is set on the info instance.

### Fixed Issues (if relevant)
1. https://github.com/magento/magento2/issues/6585

### Manual testing scenarios

1. Available on the issue description.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=14393) <br/>All committers have signed the CLA.

File: app/code/Magento/OfflinePayments/Model/Purchaseorder.php [hunk 1]
```diff
@@ -46,11 +48,29 @@ class Purchaseorder extends \Magento\Payment\Model\Method\AbstractMethod
...
+     * @return $this
+     * @throws LocalizedException
+     * @api
+     */
+    public function validate()
+    {
+        parent::validate();
+
+        if (empty($this->getInfoInstance()->getPoNumber())) {
+            throw new LocalizedException(__('Purchase order number is a required field.'));
```
Comment (Jeroenll) [inline app/code/Magento/OfflinePayments/Model/Purchaseorder.php:71]: Can you add this string to the .csv so it can be translated?

Comment (miguelbalparda) [reply app/code/Magento/OfflinePayments/Model/Purchaseorder.php:71]: Done.

Comment (centerax) [reply app/code/Magento/OfflinePayments/Model/Purchaseorder.php:71]: Thanks @miguelbalparda ! I totally forgot about this 🤦‍♂️

**Review by miguelbalparda — APPROVED** [review]

File: app/code/Magento/OfflinePayments/Model/Purchaseorder.php (hunk 1 above)
Comment (miguelbalparda) [reply app/code/Magento/OfflinePayments/Model/Purchaseorder.php:71]: All good here.

**Review by miguelbalparda — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @miguelbalparda, thank you for the review.
[ENGCOM-2382](https://jira.corp.magento.com/browse/ENGCOM-2382) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@centerax thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by magento-engcom-team** [comment]
Hi @centerax. Thank you for your contribution. 
We will aim to release these changes as part of 2.2.7. 
Please check the release notes for final confirmation.

Please, consider to port this solution to 2.3 release line.
You may use [Porting tool](https://github.com/magento/magento2/wiki/Simple-way-to-port-your-solutions-automatically) to port commits automatically.

**Comment by dnadle** [comment]
This is a bad "fix". When did Magento decide that PO# is *always* required on the payment instance? That should be set by configuration, not hard-coded.

**Comment by miguelbalparda** [comment]
I'm not that sure @dnadle, care to expand your comment? If you really think this shouldn't be in the core feel free to submit your patch with an explanation and assign it to me for review.

**Comment by centerax** [comment]
> This is a bad "fix". When did Magento decide that PO# is _always_ required on the payment instance? That should be set by configuration, not hard-coded.

The validation is performed only on the payment method Purchaseorder. I dont see how its a global validation for all payment methods as you seem to imply.

**Comment by dnadle** [comment]
OK, I take it back. We had a payment mode in production where the purchase order payment method was used with optional PO #, that was working through 2.2.6. An upgrade to 2.2.9 with this check surprised us. Sorry for the confusion.

**Comment by miguelbalparda** [comment]
Np thanks for keeping this civilized.

**Comment by ngtuan96lc** [comment]
The bug still exists on Magneto version 2.4.1
Steps: 
- click Purchase Order 
- Fill valid info and fill Purchase Order Number
- click button Purchase Order again
image
image

Root cause: purchase order number is not sent to server. 
Everyone can fix follow: Mixin file Magento_SalesRule/js/action/select-payment-method-mixin.js 

image

**Pull request merged**